File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const map = {}
        for(let i=0;i<nums.length;i++) {
            map[nums[i]] = (map[nums[i]] || 0) + 1
        }
        console.log(map)
        let heap = new MinPriorityQueue((x) => x[1]);

        for(let [num,count] of Object.entries(map)) {
            heap.enqueue([num,count])

            if(heap.size() > k) heap.dequeue()
        }

        const res = []

        for(let i=0;i<k;i++) {
            const [num,count] = heap.dequeue()
            res.push(num)
        }
        return res
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        set_ele = set(nums)
        for num in nums:
            length = 1
            if not num-1 in set_ele:
                while num+length in set_ele:
                    length += 1
                longest = max(length,longest)
        return longest